src/components/contingency-list-selector.js

```diff
@@ -18,109 +18,212 @@ import CheckboxList from './util/checkbox-list';
+                disabled={disabled}
+            >
+                <FormattedMessage id={message} />
+            </Button>
+        </Grid>
+    );
+}
 
 const ContingencyListSelector = (props) => {
     const studyUuid = decodeURIComponent(useParams().studyUuid);
 
+    const favoriteContingencyList = useSelector(
+        (state) => state[PARAM_FAVORITE_CONTINGENCY_LISTS]
+    );
+
     const [contingencyList, setContingencyList] = useState([]);
 
     const [simulatedContingencyCount, setSimulatedContingencyCount] =
         useState(0);
 
-    const [checkedContingencyListNames, setCheckedContingencyListNames] =
-        useState([]);
+    const [checkedContingencyListIds, setCheckedContingencyListIds] = useState(
+        []
+    );
+
+    const [favoriteSelectorOpen, setFavoriteSelectorOpen] = useState(false);
 
     const handleClose = () => {
         props.onClose();
     };
 
     const handleStart = () => {
-        props.onStart(checkedContingencyListNames);
+        props.onStart(checkedContingencyListIds);
     };
 
     const handleChecked = (checked) => {
-        setCheckedContingencyListNames(checked);
+        setCheckedContingencyListIds(checked);
     };
 
-    useEffect(() => {
-        if (props.open) {
-            fetchContingencyLists().then((contingencyLists) => {
-                setContingencyList(contingencyLists);
-            });
-            setCheckedContingencyListNames([]);
-        }
-    }, [props.open]);
+    const saveFavourite = (newList) => {
+        const existingLists = new Set(contingencyList.map((item) => item.id));
+        updateConfigParameter(
+            PARAM_FAVORITE_CONTINGENCY_LISTS,
+            newList.filter((item) => existingLists.has(item))
+        ).then();
+    };
 
     useEffect(() => {
         setSimulatedContingencyCount(null);
-        fetchContingencyCount(studyUuid, checkedContingencyListNames).then(
+        fetchContingencyCount(studyUuid, checkedContingencyListIds).then(
             (contingencyCount) => {
                 setSimulatedContingencyCount(contingencyCount);
             }
         );
-    }, [studyUuid, checkedContingencyListNames]);
+    }, [studyUuid, checkedContingencyListIds]);
+
+    useEffect(() => {
+        idsToObj(favoriteContingencyList, setContingencyList);
+    }, [favoriteContingencyList]);
 
     function getSimulatedContingencyCountLabel() {
         return simulatedContingencyCount != null
             ? simulatedContingencyCount
             : '...';
     }
 
+    const handleAddFavorite = () => {
+        setFavoriteSelectorOpen(true);
+    };
+
+    const handleDelete = () => {
+        const toDelete = new Set(checkedContingencyListIds);
+        saveFavourite(
+            favoriteContingencyList.filter((item) => !toDelete.has(item))
+        );
+        setCheckedContingencyListIds([]);
+    };
+
+    const removeFromFavorite = (id) => {
+        saveFavourite(favoriteContingencyList.filter((item) => item !== id));
```

Comment:
You don't reset setCheckedContingencyListIds here and remove this item.
You should factorize code here between this function and handleDelete I think.

Reply:
done